hnn_core/params.py

```diff
@@ -33,10 +34,50 @@ class Params(dict):
         The parameters json file
     """
 
-    def __init__(self, params_fname):
+    def _read_json_file(self, params_fname):
+        """Read from a .json file (default)"""
         with open(params_fname) as json_data:
             params_input = json.load(json_data)
 
+        return params_input
+
+    def _read_param_file(self, params_fname):
+        """Read from a .param file (legacy)"""
+        params_input = {}
+        with open(params_fname, 'r') as fp:
+            ln = fp.readlines()
+            for l in ln:
+                s = l.lstrip()
+                if s.startswith('#'):
+                    continue
+                sp = s.split(':')
+                if len(sp) > 1:
+                    key = sp[0].strip()
+                    value = sp[1].strip()
+                    if '.' in value or 'e' in value:
+                        try:
+                            params_input[key] = float(value)
+                        except ValueError:
+                            params_input[key] = value
+                    else:
+                        try:
+                            params_input[key] = int(value)
+                        except ValueError:
+                            params_input[key] = value
+
+        return params_input
+
+    def __init__(self, params_fname):
+        try:
+            params_input = self._read_json_file(params_fname)
+        except json.decoder.JSONDecodeError:
+            params_input = self._read_param_file(params_fname)
+
+        if len(params_input) == 0:
+            print("WARN: Failed to read any parameters from file: %s\n" %
+                  op.normpath(params_fname) +
```

Comment:
I think we should use this when reading the file name so that:

`~/Desktop/my_params/`

is the same as:

`/Users/mjas/Desktop/my_params`

Reply:
Does "this" mean op.normpath()? If the user inputs '~/something', then normpath() will fail, just as open ~/something would.

Reply:
yep this means op.normpath :) hmm ... you are right. I think I have seen something like that in MNE but what you say makes sense. 

I am wondering if we should just raise an error if no params are found. If people want to load the base parameter set, they should load it explicitly from a file to avoid any surprises

Reply:
read_params() now raises an error, but Params({}) is just a warning.